pydantic/dataclasses.py

```diff
@@ -178,7 +180,12 @@ def _process_class(
             field_info = default
             cls.__has_field_info_default__ = True
         else:
-            field_info = Field(default=default, default_factory=default_factory, **field.metadata)
+            field_info = Field(
+                default=default,
+                default_factory=default_factory,
+                **config_instance.get_field_info(field.name),
```

Comment:
Humm, I think this could lead to unexpected behaviour - `get_field_info` is not designed to return kwargs for `Field`, I think it's not used for this anywhere else.

Perhaps we could add some tests of using `get_field_info` to create `Fields` and update the docstring?

Reply:
But in theory this should all be done when creating the field as it is normally, maybe we should work out why it's not?

Reply:
This is really an issue of merging infos from config to an instance of `Field`. As a matter of fact it's the same with `create_model`

```py
from pydantic import create_model, Field

class Config:
    fields = {'a': {'description': 'descr'}}

Ad = create_model('Ad', __config__=Config, a=(str, ...))
assert Ad.schema()['properties'] == {'a': {'title': 'A', 'description': 'descr', 'type': 'string'}}

Ad = create_model('Ad', __config__=Config, a=(str, Field(...)))
assert Ad.schema()['properties'] == {'a': {'title': 'A', 'type': 'string'}}
```

@samuelcolvin I would suggest we don't include this in `1.8.1` and consider this as a classic "bug".
We can then take time to refactor calmly some stuff if needed as you suggested

Reply:
humm, I understand what you mean, but I think given that it's new since 1.8, we should try to fix it if possible.

I will look into it now.

Reply:
Turns out the same thing even happens with a vanilla pydantic model:

```py
from pydantic import Field, BaseModel

class Foo(BaseModel):
    a: str = Field(...)

    class Config:
        fields = {'a': {'description': 'descr'}}

debug(Foo.schema()['properties'])
```

I'll look into it now.